src/main/java/me/jellysquid/mods/lithium/mixin/alloc/nbt/NbtCompoundMixin.java

```diff
@@ -0,0 +1,86 @@
+package me.jellysquid.mods.lithium.mixin.alloc.nbt;
+
+import com.google.common.collect.Maps;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtElement;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyArg;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Use {@link Object2ObjectOpenHashMap} instead of {@link HashMap} to reduce NBT memory consumption and improve
+ * iteration speed.
+ *
+ * @author Maity
+ */
+@Mixin(NbtCompound.class)
+public class NbtCompoundMixin {
+
+    @Shadow
+    @Final
+    private Map<String, NbtElement> entries;
+
+    @ModifyArg(
+            method = "<init>()V",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/nbt/NbtCompound;<init>(Ljava/util/Map;)V")
+    )
+    private static Map<String, NbtElement> useFasterCollection(Map<String, NbtElement> oldMap) {
+        return new Object2ObjectOpenHashMap<>();
```

Comment:
I think this would be a great optimization, however we need to check whether anything observable depends on the iteration order of HashMap

Reply:
`HashMap` does not guarantee the iteration order. Then vanilla would have to use `LinkedHashMap`, wouldn't it?

Reply:
I suggested [this optimization](https://github.com/CaffeineMC/hydrogen-fabric/blob/1.17.x/src/main/java/me/jellysquid/mods/hydrogen/mixin/nbt/MixinNbtCompound.java) to JellySquid and she implemented it in Hydrogen. No errors were found.